### PR TITLE
[UI/Commit View] Add accordion view

### PR DESCRIPTION
**Pull request by dawidwoz**
# Problem
The current way of showing the commit view is ineffective and should be changed.

# Solution

- We added a new accordion view that can be used for anything else as it does not contain commit-specific information
- We changed the way the commit view is displayed grouping commits by the date when it was created.

It is the first step to a dedicated commit view.

The new view on screenshots:
image
image
image

### CODE REVIEW

File: src/components/accordion/AccordionItem.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    isOpen,
+    btnOnClick,
+    titleLabels
+}: {
+    data: AccordionData
+    isOpen: boolean
+    btnOnClick: () => void
+    titleLabels?: boolean
+}) {
+    const [ height, setHeight ] = useState("")
```
Comment (joglr) [inline src/components/accordion/AccordionItem.tsx:16]: I believe this state is unnecessary. Consider refactoring to a ternary expression instead, so simplify the code.

File: src/components/FileHistoryElement.tsx [hunk 2]
```diff
@@ -14,85 +18,133 @@ export function FileHistoryElement(props: props) {
 
   let fileCommits: GitLogEntry[] = []
   if (props.clickedObject.type === "blob") {
-    fileCommits = props.clickedObject.commits.map((c) => analyzerData.commits[c])
+    try {
+      fileCommits = props.clickedObject.commits.map((c) => analyzerData.commits[ c ])
+    } catch (e) {
+      console.log(e)
```
Comment (joglr) [inline src/components/FileHistoryElement.tsx:24]: Please remove this log

Comment (dawidwoz) [reply src/components/FileHistoryElement.tsx]: Solved 🙂

File: src/components/accordion/AccordionItem.tsx (hunk 1 above)
Comment (dawidwoz) [reply src/components/accordion/AccordionItem.tsx]: Solved 🙂

File: src/components/FileHistoryElement.tsx [hunk 3]
```diff
@@ -14,85 +18,125 @@ export function FileHistoryElement(props: props) {
...
+              </>
+            )
+          }) }
+        </>
+      ),
+    })
+  }
 
   return (
     <>
```
Comment (joglr) [inline src/components/FileHistoryElement.tsx:71]: This outer fragment is redundant since it only contains a single child.

File: src/components/accordion/Accordion.tsx [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+                ? !currentIdx.includes(idx)
+                : currentIdx.includes(idx)
+            }
+            titleLabels={titleLabels}
+            btnOnClick={() => {
+              btnOnClick(idx)
+              if (actionClickLabels) actionClickLabels(idx);
+            }}
+          />
+          <br />
```
Comment (joglr) [inline src/components/accordion/Accordion.tsx:66]: It seems there are some spacing issues in the commit view.
image

This could be solved with `display: flex` and using the `gap` property for specifying spacing between the entries.

File: src/components/accordion/Accordion.tsx [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        </>
+      ))}
+      <ShowMoreLabel
+        show={!showFullList && items.length > itemsCutoff}
+        items={items.slice(itemsCutoff)}
+        toggle={() => setShowFullList(!showFullList)}
+      />
```
Comment (joglr) [inline src/components/accordion/Accordion.tsx:73]: Could we have a collapse functionality as well?

**Review by joglr — CHANGES_REQUESTED** [review]
Hi Dawid, I looked into your pull request and I think the date groupings are pretty nice. 

I did not find a way to collapse the expanded commits - is this intentional?

I also left a few suggestions. Let me know what you think!

File: src/components/FileHistoryElement.tsx (hunk 3 above)
Comment (dawidwoz) [reply src/components/FileHistoryElement.tsx]: Done 🙂

File: src/components/accordion/Accordion.tsx (hunk 5 above)
Comment (dawidwoz) [reply src/components/accordion/Accordion.tsx:72]: Added 🙂

File: src/components/accordion/Accordion.tsx (hunk 4 above)
Comment (dawidwoz) [reply src/components/accordion/Accordion.tsx]: I put here just a line break `<br />` before. Now I removed it. Do you think it is better now? If not, what size should the gap be?

**Comment by dawidwoz** [comment]
> Hi Dawid, I looked into your pull request and I think the date groupings are pretty nice.
> 
> I did not find a way to collapse the expanded commits - is this intentional?
> 
> I also left a few suggestions. Let me know what you think!

@joglr I think I implemented your suggestions there is still one open question. Have a look when free 🙂

File: src/components/FileHistoryElement.tsx [hunk 6]
```diff
@@ -14,85 +19,125 @@ export function FileHistoryElement(props: props) {
...
+          <p>No commits found</p>
         </div>
       </>
     )
   }
   return (
     <>
       <div className="flex cursor-pointer justify-between hover:opacity-70">
-        <label className="label grow" htmlFor={commitHistoryExpandId}>
+        <label className="label grow">
```
Comment (joglr) [inline src/components/FileHistoryElement.tsx:102]: Removing this makes it such that clicking the title does not expand / contract the extra commit dates, so I would suggest to revert this change.

File: src/components/accordion/Accordion.tsx [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+    </ul>
+  )
+}
+
+export function ShowMoreLabel(props: ShowMoreLabelProps) {
+  if (!props.show) return null
+  return <span className="whitespace-pre text-xs font-medium opacity-70 hover:cursor-pointer" onClick={props.toggle}>+{props.items.length} day(s) more</span>
```
Comment (joglr) [inline src/components/accordion/Accordion.tsx:79]: Please reword this to "+X more day(s)"

**Review by joglr — APPROVED** [review]
I added some things that we can adress a later timer, but for now I think we should merge this.

**Comment by dawidwoz** [comment]
@joglr Thank you for merging. I'll implement your suggestions with next pull request if not done until that time.
